ray-on-gke/tpu/kuberay-tpu-webhook/main.go

```diff
@@ -708,6 +752,19 @@ func init() {
 	klog.InitFlags(nil)
 }
 
+// addPod allows next goroutine to start once the webhook PodInformer cache updates
+func (t *TPUWebhookServer) addPod(obj interface{}) {
+	// It's not guaranteed the webhook replica that admitted the Pod for this event is the same as the current caller (i.e. wg could be 0).
```

Comment:
Since we're not actually waiting for the pod event for the mutated pod, this is effectively adding some arbitrary delay to pod admission, which may be fixing the race condition with TPU_WORKER_ID.

Reply:
`AddFunc` is only triggered when a Pod with labels `"ray.io/node-type=worker,app.kubernetes.io/created-by=kuberay-operator"` is added. I can change it to check that it's a TPU Pod with the injected env vars before calling `wg.Done()`, but I wanted to err towards releasing the Wait versus continuously blocking. From my manual testing `Timed out waiting for PodInformer AddFunc` was never showing up in the logs.

Reply:
Added logic to check that the Pod in `addPod` is a TPU worker Pod before unblocking the next `Mutate` call:
04bf73c

Reply:
@andrewsykim I went ahead in [047ff5f](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/859/commits/047ff5f39ed78703f2ddfedd92e5f3605212700e) and changed it to check that the Pod admitted to the cache is the last TPU worker Pod mutated by that webhook replica before unblocking the next goroutine. We check this by adding a `lastAdmitted` var to each `TPUWebhookServer` and setting it to `<replicaIndex>-<TPU_WORKER_ID>` which are vars set for both single-host and multi-host TPUs. If the webhook Pod restarts in between Pod admission requests, the value of `lastAdmitted` will be empty and the `addPod` function will be a no-op (i.e. it won't wait for anything). The PodInformer will be initialized again and obtain an up-to-date list of Pods from the API server, so the next `Mutate` call should proceed correctly. Otherwise, each webhook `Mutate` request will wait for the PodInformer cache to update from the previous request before proceeding which should ensure unique `TPU_WORKER_ID`s even when large slice sizes result in PodInformer updates that are slower than the latency between mutating admission requests.

Reply:
From offline discussion, with [f87aec0](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/859/commits/f87aec0ae3476a886d3e4d17e7beb1c408f1ef86) we now check for `<namespace>-<RayCluster name>-<replicaIndex>-<TPU_WORKER_ID>` which should catch all the cases.